tensorflow/python/eager/function.py

```diff
@@ -2641,44 +2634,44 @@ def canonicalize_function_inputs(self, *args, **kwargs):
 
     if not kwargs:
       inputs = args
-      if self._arg_indices_to_default_values:
-        try:
-          inputs += tuple(
-              self._arg_indices_to_default_values[i]
-              for i in range(len(args), len(self._arg_names)))
-        except KeyError:
-          missing_args = [
-              self._arg_names[i]
-              for i in range(len(args), len(self._arg_names))
-              if i not in self._arg_indices_to_default_values
-          ]
+      if self._fullargspec.defaults:
+        if len(args) < self._num_req_args:
+          missing_args = self._arg_names[len(args):self._num_req_args]
           raise TypeError("{} missing required arguments: {}".format(
               self.signature_summary(), ", ".join(missing_args)))
+        inputs += tuple(
+            self._fullargspec.defaults[len(args) - self._num_req_args:])
 
       if self._fullargspec.kwonlydefaults:
         kwargs.update(self._fullargspec.kwonlydefaults)
     else:
```

Comment:
do we have a benchmark to measure function calls with kwargs?  If not, probably it's good to add one.  Maybe before landing this CL to measure the impact.

Reply:
Also, do we have a benchmark to measure function calls with kwargs?  If not, probably it's good to add one.  Maybe before landing this CL to measure the impact.

Reply:
I believe the defun_matmul_2_by_2 series calls with the `transpose_b` named arg